typescript/infra/src/govern/multisend.ts

```diff
@@ -58,28 +60,204 @@ export class SafeMultiSend extends MultiSend {
+      await this.deleteTx(tx.safeTxHash);
+    }
+  }
+
+  // Helper function to delete a single transaction
+  public async deleteTx(safeTxHash: string) {
+    const signer = this.multiProvider.getSigner(this.chain);
+    const domainId = this.multiProvider.getDomainId(this.chain);
+    const txServiceUrl = this.multiProvider.getChainMetadata(
+      this.chain,
+    ).gnosisSafeTransactionServiceUrl;
+
+    // Fetch the transaction details to get the proposer
+    const txDetailsUrl = `${txServiceUrl}/api/v1/multisig-transactions/${safeTxHash}/`;
+    const txDetailsResponse = await fetch(txDetailsUrl, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
     });
+
+    if (!txDetailsResponse.ok) {
+      console.error(`Failed to fetch transaction details for ${safeTxHash}`);
+      return;
+    }
+
+    const txDetails = await txDetailsResponse.json();
+    const proposer = txDetails.proposer;
+
+    if (!proposer) {
+      console.error(`No proposer found for transaction ${safeTxHash}`);
+      return;
+    }
+
+    // Compare proposer to signer
+    const signerAddress = await signer.getAddress();
+    if (proposer !== signerAddress) {
+      console.log(
+        `Skipping deletion of transaction ${safeTxHash} proposed by ${proposer}`,
+      );
+      return;
+    }
+    console.log(`Deleting transaction ${safeTxHash} proposed by ${proposer}`);
+
+    try {
+      // Generate the EIP-712 signature
+      const totp = Math.floor(Date.now() / 1000 / 3600); // Generate TOTP with T0=0 and Tx=3600
+      const typedData = {
+        types: {
+          EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+          ],
+          DeleteRequest: [
+            { name: 'safeTxHash', type: 'bytes32' },
+            { name: 'totp', type: 'uint256' },
+          ],
+        },
+        domain: {
+          name: 'Safe Transaction Service',
+          version: '1.0',
+          chainId: domainId,
+          verifyingContract: this.safeAddress,
+        },
+        primaryType: 'DeleteRequest',
+        message: {
+          safeTxHash: safeTxHash,
+          totp: totp,
+        },
+      };
+
+      const signature = await (signer as ethers.Wallet)._signTypedData(
+        typedData.domain,
+        { DeleteRequest: typedData.types.DeleteRequest },
+        typedData.message,
+      );
+
+      // Make the API call to delete the transaction
+      const deleteUrl = `${txServiceUrl}/api/v1/multisig-transactions/${safeTxHash}/`;
+      const res = await fetch(deleteUrl, {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          safeTxHash: safeTxHash,
+          signature: signature,
+        }),
+      });
+
+      if (res.status === 204) {
```

Comment:
It's definitely 204 every time and not any other success codes (e.g. 200)?

Reply:
that's what the spec seems to suggest, but happy to loosen up to any success code <img width="276" alt="image" src="https://github.com/user-attachments/assets/25544119-7027-47f1-872e-33ad5dca5ef9">